### PR TITLE
Update old `#index-entry-...` links

### PR DESCRIPTION
**Pull request by cfa**
Amend links with stale `#index-entry...` targets, in some cases choosing page sections rather than the equivalent index entry.

Manually tested updated links locally.

### CODE REVIEW

**Comment by coke** [comment]
Do we know why this were changed originally?

**Comment by finanalyst** [comment]
@cfa I have been working hard on this issue. I have now determined that `X<>` markup in `=head` contexts was rendered into HTML differently to `X<>` in ambient text by Documentable / `Pod::To::HTML`. I am currently working on some of this.

**Comment by coke** [comment]
What's the current status on this, @finanalyst ? Is there a ticket open tracking the related work you mention?

**Comment by finanalyst** [comment]
@coke Sorry for the considerable delay. I have been working on this issue which is also linked to other problems. Complications include
- Targets (eg. index-entry-XXX) were generated by `Documentable` in different ways depending on whether the `X<>` was included in the title, or was in the text.
- Targets with multiple entries (eg `L<display text|first entry, some description; second entry, more description>` are treated oddly, with the target being `index-entry-some_description-more_description-display_text`. 
- Targets with the form `L<C<some term>|/routine/filename>` exclude the text formatting (eg. `<code>some terms</code>` from the name of the target
- Targets with filenames containing escaped chars included things like `$SOLIDUS`
- Targets in Secondary files have their own naming form
- There are errors in the source files completely unrelated to the Documentable naming strategy

All these target names are distributed around Documentable classes, so it has been very difficult to reverse engineer the names.

I think I am making progress, and I hope to create a PR for doc-website tooling in the near future.

File: doc/Language/traps.rakudoc [hunk 1]
```diff
@@ -263,9 +263,9 @@ say A.new(x => 42).x;       # OUTPUT: «Any␤»
 leaves C<$!x> uninitialized, because the custom C<BUILD> doesn't
 initialize it.
 
-B<Note:> Consider using L<TWEAK|/language/objects#index-entry-TWEAK>
+B<Note:> Consider using L<TWEAK|/language/objects#index-entry-Methods_TWEAK>
```
Comment (finanalyst) [inline doc/Language/traps.rakudoc:266]: index-entry should stay

File: doc/Language/typesystem.rakudoc [hunk 2]
```diff
@@ -503,7 +503,7 @@ L<re-dispatching|/language/functions#Re-dispatching>.
 Classes declared with C<is hidden> also generate slightly different method
 signatures.  To facilitate re-dispatch, typical methods are automatically
 provided with an extra C<*%_> parameter that L<captures extra named
-arguments|/type/Method#index-entry-extra_named_arguments>.  Because classes
+arguments|/type/Method#index-entry-Syntax_*%___(extra_named_arguments)>.  Because classes
 declared with C<is hidden> don't participate in re-dispatch, their methods don't
```
Comment (finanalyst) [inline doc/Language/typesystem.rakudoc:507]: index-entry should stay

File: doc/Type/Any.rakudoc [hunk 3]
```diff
@@ -996,7 +996,7 @@ result, by applying a binary subroutine. It applies its argument (or first
 argument for the sub form) as an operator to all the elements in the object (or
 second argument for the sub form), producing a single result. The subroutine
 must be either an
-L<infix operator|/language/operators#index-entry-infix_operator>
+L<infix operator|/language/operators#index-entry-Language_infix_operator>
 or take two positional arguments. When using an infix operator, we must provide
```
Comment (finanalyst) [inline doc/Type/Any.rakudoc:1000]: index-entry can stay

File: doc/Type/Iterable.rakudoc [hunk 4]
```diff
@@ -72,7 +72,7 @@ itemized sublists:
 
     say ($('a', 'b'), 'c').flat;    # OUTPUT: «($("a", "b"), "c")␤»
 
-You can use the L«hyper method call|/language/operators#index-entry-methodop_>>.» to
+You can use the L«hyper method call|/language/operators#index-entry-Language_hyper_method_call_operator-hyper_method_call_operator» to
 call the L«C<.List>|/routine/List» method on all the inner itemized sublists
```
Comment (finanalyst) [inline doc/Type/Iterable.rakudoc:76]: index-entry should stay

File: doc/Type/List.rakudoc [hunk 5]
```diff
@@ -198,7 +198,7 @@ operator to flatten the tuple into an argument list, so it is equivalent to:
 
 There are other ways to list the elements of an itemized single argument. For
 example, you can L«decontainerize|/routine/<>» the argument or use the L<C<@>
-list contextualizer|/type/Any#index-entry-@_list_contextualizer>:
+list contextualizer|/type/Any#index-entry-Syntax_@_list_contextualizer>:
```
Comment (finanalyst) [inline doc/Type/List.rakudoc:201]: index-entry will be OK

File: doc/Type/List.rakudoc [hunk 6]
```diff
@@ -998,7 +998,7 @@ depends on the operator. In the functional programming world, this operation is
 generally called a L<fold|https://en.wikipedia.org/wiki/Fold_%28higher-order_function%29#Folds_on_lists>.
 With a right-associative operator it is a I<right fold>, otherwise (and usually)
 it is a I<left fold>. In Raku, you can specify the associativity of an operator
-with the L<C«is assoc»|/language/functions#index-entry-is_assoc_(trait)>.
+with the L<C«is assoc»|/language/functions#index-entry-Traits_is_assoc>.
```
Comment (finanalyst) [inline doc/Type/List.rakudoc:1002]: leave index-entry

File: doc/Type/Mu.rakudoc [hunk 7]
```diff
@@ -481,8 +481,8 @@ container to the invocant (see more details here: L<C<return-rw>|/language/contr
     method emit()
 
 Emits the invocant into the enclosing
-L<supply|/language/concurrency#index-entry-supply_(on-demand)> or
-L<react|/language/concurrency#index-entry-react> block.
+L<supply|/language/concurrency#index-entry-Reference_supply_(on-demand)> or
+L<react|/language/concurrency#react> block.
```
Comment (finanalyst) [inline doc/Type/Mu.rakudoc:486]: leave index-entry

File: doc/Type/Scalar.rakudoc [hunk 8]
```diff
@@ -23,7 +23,7 @@ In addition, C<Scalar>s delegate all method calls to the value which they
 contain. As such, C<Scalar>s are for the most part invisible. There is, however,
 one important place where C<Scalar>s have a visible impact: a C<Scalar>
 container will shield its content from
-L<flattening|/language/subscripts#index-entry-flattening_> by most Raku core list
+L<flattening|/language/subscripts#index-entry-Syntax_flattening_> by most Raku core list
 operations.
```
Comment (finanalyst) [inline doc/Type/Scalar.rakudoc:27]: leave index-entry

File: doc/Type/Sub.rakudoc [hunk 9]
```diff
@@ -54,7 +54,7 @@ L<extended identifiers|/language/syntax#Extended_identifiers>.
 X<|Syntax,trait_mod (declarator)>
 
 A C<Trait> is a sub that is applied at compile time to various objects like
-classes, routines or L<containers|/language/phasers#index-entry-Phasers__will_trait>. It
+classes, routines or L<containers|/language/phasers#index-entry-Phasers_will_trait>. It
 is declared with the C<trait_mod> declarator followed by a colon and a string
```
Comment (finanalyst) [inline doc/Type/Sub.rakudoc:58]: leave index-entry

File: doc/Type/independent-routines.rakudoc [hunk 10]
```diff
@@ -50,7 +50,7 @@ through EVAL if you activate the C<MONKEY-SEE-NO-EVAL> pragma.
 
 Please note that you can interpolate to create routine names using
 quotation, as can be seen in
-L<this example|/language/quoting#index-entry-%26_(interpolation)>
+L<this example|/language/quoting#index-entry-Syntax_&_(interpolation)>
 or
```
Comment (finanalyst) [inline doc/Type/independent-routines.rakudoc:54]: leave index-entry

**Review by finanalyst — COMMENTED** [review]
15 index-entry -> index-entry should be left as is

**Comment by finanalyst** [comment]
@cfa @coke 
- I have modified the id naming algorithm for the case when a heading contains `X<heading text|Syntax,foo>`
- This eliminates a large number of errors related to `index-entry-...` type links. 
- One example is `index-entry-qw-quote-words-quote-words` is now generated (this was the first change in @cfa PR.

Some general comments about the PR. I think that it should be 
- There are 15 changes suggested here where one `index-entry....` is replaced by another `index-entry.....`. These should now be excluded from the PR.

In general, however, I think that the changes from 'index-entry' to the heading name is better. In many cases the `X<>` markup is after the heading. Consequently when the browser hits the X anchor, the heading is placed above the browser window. When the heading is targeted the browser puts the heading in a visible position.

The changes to the production web-site have yet to be approved. So the new index-entry names can be seen in new-raku.finanalyst.org

**Comment by coke** [comment]
@cfa  - can you rebase this and address @finanalyst's comments? Would love to get it merged finally!

**Comment by coke** [comment]
Instead of trying to apply this, will work off the error report referenced in https://github.com/Raku/doc/issues/4476

Thanks, @cfa, for giving this a shot, I'm sorry we didn't get it merged in time. :(

**Pull request closed without merging**